Havtorn/Source/Editor/AuthoringTools/ScriptTool.cpp

```diff
@@ -191,29 +189,33 @@ namespace Havtorn
 
 	void CScriptTool::OnDisable()
 	{
+		CloseScript();
 	}
 
-	void CScriptTool::OpenScript(SScript* script)
+	void CScriptTool::OpenScript(SEditorAssetRepresentation* asset)
 	{
-		CurrentScript = script;
+		CurrentScriptRepresentation = asset;
+		CurrentScript = GEngine::GetAssetRegistry()->RequestAssetData<HexRune::SScript>(SAssetReference(asset->DirectoryEntry.path().string()), CAssetRegistry::EditorManagerRequestID);
 		SetEnabled(true);
-
-		//LoadGUIElements();
-		//GUI::OpenScript(GUINodes, GUILinks);
 	}
 
-	void CScriptTool::CloseScript()
+	void CScriptTool::SaveScript()
 	{
-		SetEnabled(false);
+		SScriptFileHeader asset;
+		asset.Name = UGeneralUtils::ExtractFileBaseNameFromPath(CurrentScriptRepresentation->DirectoryEntry.path().string());
+		asset.Script = CurrentScript;
 
-		//LoadGUIElements();
-		//GUI::CloseScript(GUINodes, GUILinks);
+		Manager->GetResourceManager()->CreateAsset(CurrentScriptRepresentation->DirectoryEntry.path().parent_path().string() + "\\", asset);
```

Comment:
Not neccesary to fix now- We just dont want to forget we wanna sort this one out soon
```suggestion
		Manager->GetResourceManager()->CreateAsset(CurrentScriptRepresentation->ParentPath(), asset);
```